core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -323,10 +332,10 @@ public void watch() {
     }
   }
 
-  private void onEvent(EventType eventType, Path childPath, int count, Path rootPath)
+  private void onEvent(EventType eventType, boolean isDirectory, Path childPath, int count, Path rootPath)
       throws IOException {
-    logger.debug("-> {} [{}]", eventType, childPath);
-    listener.onEvent(new DirectoryChangeEvent(eventType, childPath, count, rootPath));
+    logger.debug("-> {} [{}, {}]", eventType, isDirectory, childPath);
```

Comment:
Just to make this log more readable I'd do something like `"-> {} [{}] (isDirectory: {})"`

Reply:
done